test_runner/pg_clients/test_pg_clients.py

```diff
@@ -0,0 +1,51 @@
+import os
+import shutil
+import subprocess
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from urllib.parse import urlparse
+
+import pytest
+from fixtures.neon_fixtures import RemotePostgres
+
+
+@pytest.mark.remote_cluster
+@pytest.mark.parametrize(
+    "client",
+    [
+        "csharp/npgsql",
+        "java/jdbc",
+        "python/asyncpg",
+        pytest.param("python/pg8000", marks=pytest.mark.xfail(reason="Doesn't support SNI")),
```

Comment:
I don't understand the error we get `scramp.core.ScramException: The channel_binding parameter must be None if channel binding is not required.` but it seems like it's pointing at a bug in our scram server-side protocol maybe? cc @funbringer

Reply:
Here actually I'm not sure if it's a bug in our code or in `pg8000`.
If I comment out this block:
https://github.com/tlocke/pg8000/blob/1.29.1/pg8000/core.py#L264-L267 it starts to work for me (creating custom `ssl_context` with `request_ssl=False` doesn't really work)

Reply:
Yeah looks like the library assumes that if ssl is requested then channel binding is required, which is not the case for our proxy. Let's leave a comment to clarify.

Reply:
I'm busy with some other stuff, but with my superficial understanding this should be fixed on the proxy's side. I'll take a closer look later.